nav2_smac_planner/src/smac_planner_hybrid.cpp

```diff
@@ -340,8 +340,23 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
 #endif
 
   // Smooth plan
-  if (num_iterations > 1 && plan.poses.size() > 6) {
-    _smoother->smooth(plan, costmap, time_remaining);
+  // Check if there are any cusps and if so, split path to avoid smoothing cusps
+  if (num_iterations > 1) {
+    std::vector<unsigned int> cusps_id = findDirectionChange(plan);
+    if (cusps_id.size() < 3 && plan.poses.size() > 6) {
+        _smoother->smooth(plan, costmap, time_remaining);
+    } else {
+      for (unsigned int i = 1; i < cusps_id.size(); ++i) {
+        if (cusps_id[i] - cusps_id[i - 1] > 6) {
```

Comment:
This is the right idea, but I think this should be in the smoother class's `smooth` instead of the planner implementation. It's a specific niche of this specific smoother, not a planner-level concern.

Reply:
I've moved this to the smoother class' smooth as you suggested